web/src/Room.tsx

```diff
@@ -117,6 +117,37 @@ export default function Room() {
   const [votedToSkip, setVotedToSkip] = useState(false);
   const [skipVotes, setSkipVotes] = useState(0);
 
+  function sanitizeSlug(input: string): string {
+    // Slugs use modified Base64 with "-_" instead of "+/"
+    const slugRegex = /^[\w\-]+$/;
+
+    // https://stackoverflow.com/questions/19377262/regex-for-youtube-url
+    // See https://regex101.com/r/vHEc61/1 for test cases.
+    const ytRegex = new RegExp(
+      /^((?:https?:)?\/\/)?/.source + /((?:www|m)\.)?/.source
+      + /((?:youtube(?:-nocookie)?\.com|youtu.be))/.source
+      + /(\/(?:[\w\-]+\?v=|embed\/|live\/|v\/)?)/.source
+      + /([\w\-]+)/.source // The slug is contained in the 5th group
+      + /(\S+)?$/.source
+    );
```

Comment:
Don't use a regular expression for this. Instead, use the [`URL`](https://developer.mozilla.org/en-US/docs/Web/API/URL/URL) API. You should be able to use it to parse the URL and extract the query parameter.

Reply:
Ah much cleaner now. Lmk if this looks right.